pyfr/integrators/dual/pseudo/pseudocontrollers.py

```diff
@@ -74,11 +74,18 @@ def conv_mon(self, i, minniters):
 
         self.pseudostepinfo.append((self.npseudosteps, i + 1, resid))
 
-        return resid and max(resid[1:]) < self._pseudo_residtol
+        converged = all(
+            r <= t for r, t in zip(resid, self._pseudo_residtol.values())
```

Comment:
If _pseudo_residtol is a list you can do converged = resid < tuple(self._pseudo_residtol).